pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,512 @@
+	setAnnotation(wrappedObject, originWdsAnnotation, c.wdsName)
+
+	for _, destination := range placementDecision.Spec.Destinations { // TODO need to revisit the destination struct and see how to use it properly
+		if err := c.createOrUpdateWrappedObject(ctx, destination.ClusterId, wrappedObject); err != nil {
+			return fmt.Errorf("failed to propagate wrapped object '%s' to all required WECs - %w", wrappedObject.GetName(), err)
+		}
+	}
+
+	return nil
+}
+
+func (c *genericTransportController) getObjectsFromWDS(ctx context.Context, placementDecision *v1alpha1.PlacementDecision) ([]*unstructured.Unstructured, error) {
+	objectsToPropagate := make([]*unstructured.Unstructured, 0)
+	// add cluster-scoped objects to the 'objectsToPropagate' slice
+	for _, clusterScopedObject := range placementDecision.Spec.Workload.ClusterScope {
+		if clusterScopedObject.ObjectNames == nil {
+			continue // no objects from this gvr, skip
+		}
+		gvr := schema.GroupVersionResource{Group: clusterScopedObject.Group, Version: clusterScopedObject.Version, Resource: clusterScopedObject.Resource}
+		for _, objectName := range clusterScopedObject.ObjectNames {
+			object, err := c.wdsDynamicClient.Resource(gvr).Get(ctx, objectName, metav1.GetOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("failed to get required cluster-scoped object '%s' with gvr %s from WDS - %w", objectName, gvr, err)
+			}
+			objectsToPropagate = append(objectsToPropagate, cleanObject(object))
+		}
+	}
+	// add namespace-scoped objects to the 'objectsToPropagate' slice
+	for _, namespaceScopedObject := range placementDecision.Spec.Workload.NamespaceScope {
+		gvr := schema.GroupVersionResource{Group: namespaceScopedObject.Group, Version: namespaceScopedObject.Version, Resource: namespaceScopedObject.Resource}
+		for _, objectsByNamespace := range namespaceScopedObject.ObjectsByNamespace {
+			if objectsByNamespace.Names == nil {
+				continue // no objects from this namespace, skip
+			}
+			for _, objectName := range objectsByNamespace.Names {
+				object, err := c.wdsDynamicClient.Resource(gvr).Namespace(objectsByNamespace.Namespace).Get(ctx, objectName, metav1.GetOptions{})
+				if err != nil {
+					return nil, fmt.Errorf("failed to get required namespace-scoped object '%s' with gvr '%s' from WDS - %w", objectName, gvr, err)
+				}
+				objectsToPropagate = append(objectsToPropagate, cleanObject(object))
+			}
+		}
+	}
+
+	return objectsToPropagate, nil
+}
+
+func (c *genericTransportController) createOrUpdateWrappedObject(ctx context.Context, namespace string, wrappedObject *unstructured.Unstructured) error {
+	existingWrappedObject, err := c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Get(ctx, wrappedObject.GetName(), metav1.GetOptions{})
+	if err != nil {
+		if !strings.HasSuffix(err.Error(), notFoundErrorSuffix) { // if object is not there, we need to create it. otherwise report an error.
```

Comment:
Why not use `errors.IsNotFound(err)` here?

Reply:
good point. done.